File: app/src/main/java/net/farnetwork/latihanespresso/LoginActivity.java
package net.farnetwork.latihanespresso;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity {

    EditText textUsername;
    EditText textPassword;
    Button btnLogin;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        loadView();
        init();
    }

    private void loadView() {
        textUsername = findViewById(R.id.textUsername);
        textPassword = findViewById(R.id.textPassword);
        btnLogin = findViewById(R.id.btnLogin);
    }

    private void init() {
        btnLogin.setOnClickListener(v -> {
            String username = textUsername.getText().toString();
            String password = textPassword.getText().toString();

            if (!validateLogin(username, password))
                return;

            doLogin(username, password);
        });
    }

    private void doLogin(String username, String password) {
        if (username.equals("user") && password.equals("password")) {
            Toast.makeText(
                    this,
                    getString(R.string.welcomeMessage, username),
                    Toast.LENGTH_SHORT
            ).show();
        } else {
            Toast.makeText(this, R.string.wrongUsernamePassword, Toast.LENGTH_SHORT).show();
        }
    }

    private boolean validateLogin(String username, String password) {
        boolean isValid = true;

        if (username.isEmpty()) {
            isValid = false;
            textUsername.setError(getString(R.string.emptyUsername));
            textUsername.setFocusable(true);
        }
        if (password.isEmpty()) {
            isValid = false;
            textPassword.setError(getString(R.string.emptyPassword));
        }

        return isValid;
    }
}

File: app/src/main/java/net/farnetwork/latihanespresso/kotlin/LoginActivity.kt
package net.farnetwork.latihanespresso.kotlin

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_login.*
import net.farnetwork.latihanespresso.R

class LoginActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        init()
    }

    private fun init() {
        btnLogin.setOnClickListener {
            val username: String = textUsername.text.toString()
            val password: String = textPassword.text.toString()
            if (!validateLogin(username, password)) return@setOnClickListener
            doLogin(username, password)
        }
    }

    private fun doLogin(username: String, password: String) {
        if (username == "user" && password == "password") {
            Toast.makeText(
                    this,
                    getString(R.string.welcomeMessage, username),
                    Toast.LENGTH_SHORT
            ).show()
        } else {
            Toast.makeText(this, R.string.wrongUsernamePassword, Toast.LENGTH_SHORT).show()
        }
    }

    private fun validateLogin(username: String, password: String): Boolean {
        var isValid = true
        if (username.isEmpty()) {
            isValid = false
            textUsername.error = getString(R.string.emptyUsername)
            textUsername.isFocusable = true
        }
        if (password.isEmpty()) {
            isValid = false
            textPassword.error = getString(R.string.emptyPassword)
        }
        return isValid
    }
}
